cubed/storage/zarr.py

```diff
@@ -20,18 +21,21 @@ def __init__(
         dtype: T_DType,
         chunks: T_RegularChunks,
         path: Optional[str] = None,
+        storage_options: Optional[Dict[str, Any]] = None,
         **kwargs,
     ):
         """Create a Zarr array lazily in memory."""
         # use an empty in-memory Zarr array as a template since it normalizes its properties
         template = zarr.empty(
             shape, dtype=dtype, chunks=chunks, store=zarr.storage.MemoryStore()
         )
+        if storage_options:
+            s3 = s3fs.S3FileSystem(**storage_options)
+            store = s3fs.S3Map(root=store, s3=s3, check=False)
```

Comment:
This code is S3-specific, which we don't want to hardcode. In fact, I think you don't need any changes in zarr.py at all - since the `storage_options` from the spec will be passed straight through to [`zarr.open_array`](https://zarr.readthedocs.io/en/stable/api/creation.html#zarr.creation.open_array). Does that work in your case?

Reply:
You are right. They can be passed through **kwargs. 
However kwargs were not passed to `zarr.open_array` inside `LazyZarrArray.open()` so I made that modification.